### PR TITLE
Add test for js files.

### PR DESCRIPTION
**Pull request by jpmedley**
https://github.com/google/WebFundamentals/issues/4017

### CODE REVIEW

File: gulp-tasks/test.js [hunk 1]
```diff
@@ -242,6 +242,26 @@ function findBadMDExtensions() {
 }
 
 /**
+ * Chec to ensure there an o JavaSCript files in the content/ folder.
```
Comment (petele) [inline gulp-tasks/test.js]: Please fix the description here, missing a few letters ;)

File: gulp-tasks/test.js [hunk 2]
```diff
@@ -242,6 +242,26 @@ function findBadMDExtensions() {
...
+ *
+ * Note: the promise will always be fulfilled, even if it fails.
+ *
+ * @return {promise} An empty promsie
```
Comment (petele) [inline gulp-tasks/test.js]: Promise in `{}` should be capitalized

File: gulp-tasks/test.js [hunk 3]
```diff
@@ -242,6 +242,26 @@ function findBadMDExtensions() {
...
+ */
+ function findJavaSCriptFiles() {
```
Comment (petele) [inline gulp-tasks/test.js]: Change to `findJavaScriptFiles()`

File: gulp-tasks/test.js [hunk 4]
```diff
@@ -242,6 +242,26 @@ function findBadMDExtensions() {
...
+  return new Promise(function(resolve, reject) {
+    var contentPath = GLOBAL.WF.src.content;
```
Comment (petele) [inline gulp-tasks/test.js]: Need to add support for `testPath`, which is used when only doing a single language.

You can probably do something like:
```
var contentPath = GLOBAL.WF.src.content;
if (GLOBAL.WF.options.testPath) {
  contentPath = GLOBAL.WF.options.testPath;
}

File: gulp-tasks/test.js [hunk 5]
```diff
@@ -242,6 +242,26 @@ function findBadMDExtensions() {
...
+    var patterns = ['**/*.js', '!**/_code/*.js', '!**/event-map.js'];
+    var files = wfHelper.getFileList(contentPath, patterns);
+    files.forEach(function(filename) {
+      summary.files++;
+      logError(filename.filePath, '\n\tJS files must be in a directory named /_code.')
```
Comment (petele) [inline gulp-tasks/test.js]: Be consistent in logging, don't use the `\n\t`, instead I'd recommend: 'JavaScript files are not permitted.'

File: gulp-tasks/test.js [hunk 6]
```diff
@@ -520,7 +540,8 @@ gulp.task('test', function(callback) {
       return Promise.all([
         testAllYaml(),
         testAllMarkdown(),
-        findBadMDExtensions()
+        findBadMDExtensions(),
+        findJavaSCriptFiles()
```
Comment (petele) [inline gulp-tasks/test.js]: Be sure to update once you've renamed the function.

File: gulp-tasks/test.js (hunk 5 above)
Comment (jpmedley) [inline gulp-tasks/test.js]: Am I misunderstanding what `_code`/ is for?

The `\n\t` helps the readability of the output a bit. Can we keep it?

**Comment by jpmedley** [comment]
@petele 
Your request for line 258 isn't finished because I have two questions.

Am I misunderstanding what _code/ is for?

The \n\t helps the readability of the output a bit. Can we keep it?

File: gulp-tasks/test.js (hunk 5 above)
Comment (petele) [inline gulp-tasks/test.js]: I'd prefer to drop it for consistency right now as we're not using it anywhere else. 

As for the `_code/` directory, all JS files checked into the Google Source Repo need to be reviewed by someone with JS readability, which makes that CL harder to get approved. I'd rather discourage people from doing it unless it's a special case.

**Comment by petele** [comment]
I'd prefer to drop it for consistency right now as we're not using it anywhere else. 

As for the `_code/` directory, all JS files checked into the Google Source Repo need to be reviewed by someone with JS readability, which makes that CL harder to get approved. I'd rather discourage people from doing it unless it's a special case.

**Comment by jpmedley** [comment]
PTAL

**Pull request merged**